### PR TITLE
GLTFLoader: Fix bug when loading external WebP texture

### PR DESCRIPTION
**Pull request by gkjohnson**
Related issue: --

**Description**

When loading an external WebP texture file with GLTFLoader and a texture loader is not retrieved with `manager.getHandler` the loader variable is set to null resulting in an exception. To address it I've changed the logic to mirror what happens in [the non-WebP loadTexture function](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/GLTFLoader.js#L2435-L2449). I found this when looking into a privately shared set of data provided to me to debug an issue with the [3DTilesRendererJS](https://github.com/NASA-AMMOS/3DTilesRendererJS) package so unfortunately I cannot share a file for testing but hopefully the logic issue is clear.

cc @donmccurdy

### CODE REVIEW

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -763,7 +763,19 @@ THREE.GLTFLoader = ( function () {
...
-		var loader = source.uri ? parser.options.manager.getHandler( source.uri ) : parser.textureLoader;
+
+		var loader;
+		if ( source.uri ) {
+
+			loader = parser.options.manager.getHandler( source.uri );
+
+		}
+
+		if ( ! loader ) {
```
Comment (mrdoob) [inline examples/js/loaders/GLTFLoader.js:774]: I think `} else {` would be better here.

Comment (gkjohnson) [reply examples/js/loaders/GLTFLoader.js:774]: Then the logic would be same as before. The intent here is to handle the case `getHandler` returns a `null` value which is not tested in the above condition.

Comment (mrdoob) [reply examples/js/loaders/GLTFLoader.js:774]: Ah I see... May be good to add a quick comment then?

Comment (gkjohnson) [reply examples/js/loaders/GLTFLoader.js:774]: It's the same pattern as used in [this part of the code in the same file](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/loaders/GLTFLoader.js#L2435-L2449). Should I add a comment in both places?

Comment (mrdoob) [reply examples/js/loaders/GLTFLoader.js:774]: The main issue is the `! loader`. I'm not a fan of variables that can be `undefined` or `null`.

How about this?

```js
		var loader = parser.textureLoader;

		if ( source.uri ) {

			var handler = parser.options.manager.getHandler( source.uri );
			if ( handler !== null ) loader = handler;

		}
```

Comment (gkjohnson) [reply examples/js/loaders/GLTFLoader.js:774]: I'm fine with it -- I was just trying to follow the already established pattern in the file. Do you want me to change both places?

Comment (mrdoob) [reply examples/js/loaders/GLTFLoader.js:774]: Yes please 🙏

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -2367,17 +2374,14 @@ THREE.GLTFLoader = ( function () {
 
 		source = json.images[ textureDef.source ];
 
-		var loader;
+		var loader = textureExtensions[ EXTENSIONS.MSFT_TEXTURE_DDS ]
+			? parser.extensions[ EXTENSIONS.MSFT_TEXTURE_DDS ].ddsLoader
+			: this.textureLoader;
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js:2379]: This looks like a mistake during a rebase — support for `MSFT_texture_dds` was removed in https://github.com/mrdoob/three.js/pull/21271 and shouldn't be referenced here.

Comment (gkjohnson) [reply examples/js/loaders/GLTFLoader.js]: Oh you're right -- thanks for the catch. I just updated it. Mind taking another look? I removed some unused variables in the function (specifically `parser` and `textureExtensions`) and simplified the definition of `source`, too.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**